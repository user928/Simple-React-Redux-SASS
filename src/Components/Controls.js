import React from 'react';
import { connect, Provider } from 'react-redux';
import store from '../Redux/store';
import PropTypes from 'prop-types';
import { decrementNum, incrementNum, resetCounter } from '../Redux/actionCreators/counterActions';

export class _Controlls extends React.Component {
	static propTypes = {
		increment: PropTypes.func,
		decrement: PropTypes.func,
		resetCount: PropTypes.func,
		resetLabel: PropTypes.string,
		incrementLabel: PropTypes.string,
		decrementLabel: PropTypes.string,
	};

	render() {
		return (
			<div className="controls">
				<div className="increment" onClick={this.props.increment}>
					<span>+</span>
				</div>

				<div className="reset" onClick={this.props.resetCount}>
					<span>Reset this</span>
				</div>

				<div className="decrement" onClick={this.props.decrement}>
					<span>-</span>
				</div>
			</div>
		);
	}
}

const mapDispatchToProps = dispatch => ({
	increment: () => dispatch(incrementNum()),
	decrement: () => dispatch(decrementNum()),
	resetCount: () => dispatch(resetCounter()),
});

const Controlls = connect(
	null,
	mapDispatchToProps
)(_Controlls);

export default class extends React.Component {
	render() {
		return (
			<Provider store={store}>
				<Controlls {...this.props} />
			</Provider>
		);
	}
}
