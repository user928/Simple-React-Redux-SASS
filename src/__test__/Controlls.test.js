import Controls from '../Components/Controls.js';


describe('Controls component', () => {

	it('renders correctly', () => {
		const wrapper = render(
			<Controls/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('find reset txt', () => {
		const wrapper = mount(
			<Controls />
		);
		const resetTxt = wrapper.find('.reset').childAt(0).text();
		expect(resetTxt).toBe('Reset');
	});

	it('click on increment once', () => {
		const spy = sinon.spy();
		const wrapper = mount(
			<Controls increment={spy} />
		);

		wrapper.find('.increment').simulate('click');
		expect(spy.calledOnce).toBe(true);
	});

});