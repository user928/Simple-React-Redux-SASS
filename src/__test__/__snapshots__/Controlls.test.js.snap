// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders correctly 1`] = `
<div
  class="controls"
>
  <div
    class="increment"
  >
    <span>
      +
    </span>
  </div>
  <div
    class="reset"
  >
    <span>
      Reset
    </span>
  </div>
  <div
    class="decrement"
  >
    <span>
      -
    </span>
  </div>
</div>
`;
